### PR TITLE
Update to use credo rather than dogma

### PR DESCRIPTION
**Pull request by hassox**
Also updates poison.

Credo seems to pick up a _lot_ more than dogma and that's for sure something I need.

### CODE REVIEW

File: lib/guardian.ex [hunk 1]
```diff
@@ -35,9 +35,9 @@ defmodule Guardian do
   The resource will be run through the configured serializer
   to obtain a value suitable for storage inside a JWT.
   """
-  @spec encode_and_sign(any) :: { :ok, String.t, Map } |
```
Comment (doomspork) [inline lib/guardian.ex]: Credo catches these?  Awesome!

Comment (hassox) [inline lib/guardian.ex]: there were so many :|

**Comment by doomspork** [comment]
Looks good to me @hassox :+1:

**Comment by aaronrenner** [comment]
Good find @hassox. It looks like the [default config](https://github.com/rrrene/credo/blob/master/.credo.exs) doesn't lint any of the tests. I'm fine with not linting the tests, but I just wanted to point that out.

**Comment by hassox** [comment]
@aaronrenner do you happen to know the Magic Config to get it to lint the tests? I think we should probably include those

**Comment by aaronrenner** [comment]
It looks like you create a `.credo.exs` file in either `config/` or the root folder, and then change the `files:` > `included:` array. https://github.com/rrrene/credo/blob/master/.credo.exs#L20. I'm not sure if you have to copy the entire config file into your project or if it merges with the default settings. I'd be happy to take a look at it in the morning if you want, but right now I'm off to bed.

**Comment by hassox** [comment]
Thanks for the pointer. I'll pick it up on the train into work in the am

**Pull request merged**